platform/lib/build/samplesBuilder.js

```diff
@@ -258,6 +272,57 @@ class SamplesBuilder {
     });
   }
 
+  /**
+   * Adds a sample to the sitemap object for the playground to then render
+   * a list of available samples
+   * @param {Vinyl} sample       The file from which the sample is parsed
+   * @param {Object} parsedSample The parsed sample
+   */
+  _addToSitemap(sample, parsedSample) {
+    const format = this._getSampleFormat(parsedSample);
+    const formatCategories = this._sitemap[format] || {};
+
+    const category = this._getCategory(sample);
+    const categorySamples = formatCategories[category] || {
+      'name': parsedSample.category().publicName,
+      'examples': []
+    };
+    categorySamples.examples.push({
+      'title': parsedSample.document.title,
+      'url': `${config.getHost(config.hosts.preview)}` +
+          this._getSourceRoute(sample),
+    });
+
+    formatCategories[category] = categorySamples;
+    this._sitemap[format] = formatCategories;
+  }
+
+  /**
+   * Takes what has been saved to this._sitemap and adds a sitemap.json to
+   * the gulp stream that is usable by the playground
+   * @type {Vinyl}
+   */
+  async _generateSitemap() {
+    for (const format of Object.keys(this._sitemap)) {
```

Comment:
FYI `for (const [key, value] of Object.entries(this._sitemap)) {`

Reply:
Nice, thanks!